src/Components/Shared/src/WebEventData.cs

```diff
@@ -57,6 +57,18 @@ private WebEventData(int browserRendererId, ulong eventHandlerId, EventFieldInfo
 
         public EventArgs EventArgs { get; }
 
+        [DynamicDependency(JsonSerialized, typeof(ChangeEventArgs))]
```

Comment:
I imagine the need to do this would go away once we have the JSON code generator. In the meanwhile, I ran the BasicTestApp locally and all of the scenarios worked. Working on getting tests sorted next.

Reply:
I'm OK with this being here, especially if it's temporary, but it is strange.

It's not as if the developer manually create some chain of calls that lead here. It's not really a *dynamic* dependency. In reality the framework hard-codes the idea that the event dispatch code paths are reachable, so it's very arbitrary that this particular method is where we put the rules about retaining `ClipboardEventArgs` et al. It would be potentially difficult to find these attributes later if we wanted, since there's no clear logical deduction a developer could make to realise they would be here.

If there was an alternative to annotate the eventargs subclasses directly to say "never trim this" I think that would be a lot clearer. Don't know if there is such a mechanism like the following:

```cs
[NeverTrim] public class ClipboardEventArgs : EventArgs { ... }
```

Reply:
> If there was an alternative to annotate the eventargs subclasses directly to say "never trim this" I think that would be a lot clearer.

AFAIK, there isn't a non-XML way to do this. That said, I was able to modify the code to avoid any trimmer specific by invoking the JSON serialization inline.